src/extractor/guidance/mergable_road_detector.cpp

```diff
@@ -317,6 +317,59 @@ bool MergableRoadDetector::HaveSameDirection(const NodeID intersection_node,
 
     const auto connect_again = (coordinates_to_the_left.back() == coordinates_to_the_right.back());
 
+    // Tuning parameter to detect and don't merge roads close to circular shapes
+    // if the area to squared circumference ratio is between the lower bound and 1/(4π)
+    // that correspond to isoperimetric inequality 4πA ≤ L² or lower bound ≤ A/L² ≤ 1/(4π).
+    // The lower bound must be larger enough to allow merging of square-shaped intersections
+    // with A/L² = 1/16 or 78.6%
+    // The condition suppresses roads merging for intersections like
+    //             .  .
+    //           .      .
+    //       ----        ----
+    //           .      .
+    //             .  .
+    // but will allow roads merging for intersections like
+    //           -------
+    //          /       \ 
+    //      ----         ----
+    //          \       /
+    //           -------
+    const auto constexpr CIRCULAR_POLYGON_ISOPERIMETRIC_LOWER_BOUND = 0.85 / (4 * M_PI);
+    if (connect_again && coordinates_to_the_left.front() == coordinates_to_the_left.back())
```

Comment:
Can we also do this, if the road entering / exiting the circle has the same name? There are many places that offer circular shapes and the same name for the circle as well as a single crossing road.
In that case, we would probably have to append the coordinates to the right to the coordinates to the left.

Reply:
I added a check for  polygons combined  of `coordinates_to_the_left` and reversed `coordinates_to_the_right` if `coordinates_to_the_left` is not a closed polygon, but it is does not show any new intersection.  I would keep only checking closed `coordinates_to_the_left`